frontend/src/components/Graph.tsx

```diff
@@ -176,10 +179,13 @@ export default class Graph extends React.Component<GraphProps, GraphState> {
           // deviating from the explicit layout provided by dagre.
           if (finalSegment) {
             const destinationNode = graph.node(edgeInfo.w);
+            if (destinationNode === undefined) {
+              continue;
+            }
 
             // Placeholder nodes never need adjustment because they always have only a single
             // incoming edge.
-            if (!destinationNode.isPlaceholder) {
+            if (destinationNode && !destinationNode.isPlaceholder) {
```

Comment:
This can be removed, especially if the `destinationNode === undefined` is kept

Reply:
done